broker/broker-adapter/src/main/java/org/codice/ddf/broker/security/EncryptionCodec.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.broker.security;
+
+import java.util.Map;
+
+import org.apache.activemq.artemis.utils.SensitiveDataCodec;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+
+import ddf.security.encryption.EncryptionService;
+
+public class EncryptionCodec implements SensitiveDataCodec<String> {
+
+    EncryptionService encryptionService;
+
+    @Override
+    public String decode(Object o) throws Exception {
+        if (getEncryptionService() == null) {
```

Comment:
Is this a valid return? It seems not obvious to me - why not throw an exception that says the `encryptionService` is `null`. It just seems to me that we could have failed authentication way down the line because the username and password didn't get decrypted correctly and it would be hard to trace back